services/payments/src/simcore_service_payments/api/rest/_acknowledgements.py

```diff
@@ -19,34 +29,82 @@
 
 
 async def on_payment_completed(
-    payment_id: PaymentID, ack: AckPayment, session: SessionData
+    transaction: PaymentsTransactionsDB, rut_api: ResourceUsageTrackerApi
 ):
+    assert transaction.completed_at is not None  # nosec
+    assert transaction.initiated_at < transaction.completed_at  # nosec
+
+    _logger.debug("TODO next PR Notify front-end of payment -> sio ")
+
+    with log_context(
+        _logger,
+        logging.INFO,
+        "Top-up %s credits for %s in RUT",
+        f"{transaction.osparc_credits}",
+        f"{transaction.payment_id=}",
+    ):
+        credit_transaction_id = await rut_api.create_credit_transaction(
+            product_name=transaction.product_name,
+            wallet_id=transaction.wallet_id,
+            wallet_name="id={transaction.wallet_id}",
+            user_id=transaction.user_id,
+            user_email=transaction.user_email,
+            osparc_credits=transaction.osparc_credits,
+            payment_transaction_id=transaction.payment_id,
+            created_at=transaction.completed_at,
+        )
+
     _logger.debug(
-        "payment completed: %s",
-        f"{payment_id=}, {ack.success=}, {ack.message=}, {session.username=}",
+        "RUT response to %s was %s",
```

Comment:
S: using acronyms in text? not sure that is a very good practice. I think the log should be as clear as possible.